compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -971,7 +972,7 @@ impl<'tcx> LayoutOfHelpers<'tcx> for CodegenCx<'_, 'tcx> {
         if let LayoutError::SizeOverflow(_) = err {
             self.sess().emit_fatal(Spanned { span, node: err })
         } else {
-            span_bug!(span, "failed to get layout for `{}`: {}", ty, err)
+            self.tcx.sess.emit_fatal(ssa_errors::FailedToGetLayout { span, ty, err })
         }
```

Comment:
Could we go without the branch in the 3 `handle_layout_err` methods, and always emit the same diagnostic struct?

Reply:
Perhaps...

Or now I'm thinking, maybe the more correct thing to do here is:

```
        if let LayoutError::SizeOverflow(_) | LayoutError::Cycle = err {
            self.tcx.sess.emit_fatal(ssa_errors::FailedToGetLayout { span, ty, err })
        } else {
            span_bug!(span, "failed to get layout for `{}`: {}", ty, err)
        }
```

since @oli-obk only really made changes for `LayoutError::Cycle`, the rest is still probably a bug?